app/src/main/java/com/poupa/vinylmusicplayer/ui/activities/SettingsActivity.java

```diff
@@ -139,6 +141,24 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
 
     public static class SettingsFragment extends ATEPreferenceFragmentCompat implements SharedPreferences.OnSharedPreferenceChangeListener {
 
+        ActivityResultLauncher sharedPreferencesImporter;
+
+        @Override
+        public void onCreate(@Nullable Bundle savedInstanceState) {
```

Comment:
> Hmm, in the case of manual change of primary/accent color, an explicit `Activity.recreate` is called to force redraw with new theme. cf `SettingsActivity.onColorSelection` And afaik, this will also let ATH library (containing ThemeStore) to update their internal stuffs.
> 
> Have you tried this approach?

With this change it works now